📝 Add the Documentation for the StartEvent (#3)
atlas-engine/ContributionGuidelines: docs/getting_started/bpmn_elements/start_event.md

This can come in handy if you want to define multiple starting points
inside your process. You can also use multiple StartEvents to define
completely different Flows inside a single Process.

![multiple_start_events](images/start_event/multiple_start_events.svg)

In this process you can find two different StartEvents. In this example, the
Task which executes some prerequisites should only be executed, if the Process
is started the first time.
This is solved by using multiple StartEvents. The StartEvent named _Initial Startup_
indicates, that the Process only starts here, if it needs to do some initial work
first.
The other StartEvent named _Normal Startup_ indicates, that the Process should
start here, if its already initialized.

On the AtlasEngine, the different StartEvents are always defined by their
Ids. As you can see in the notes above, the _Initial Startup_ StartEvent
has the Id `initial_start` whereas the _Normal Startup_ StartEvent has the
Id `normal_start`.
You can use this Id to tell the AtlasEngine, which StartEvent should be used
when starting a new ProcessInstance.

## Message Start Event

This variant of a StartEvent waits until a defined Message was received.

> If the Message was catched, the Process will be started at the MessageStartEvent.

Reviewer: ```suggestion
This type of StartEvent starts a process automatically, as soon as the message attached to the Start Event was received by the Atlas Engine.
```

Each MessageStartEvent has to be configured with a _Message_.
* A message contains of an `id` and a `name`. Internally, the MessageId is used to
  determine, if the target Message was received.

> * You can easily configure the message inside the BpmnStudio.

Reviewer: Das ist leider genau falsch rum. :upside_down_face: 

Es wird anhand des Message Namen bestimmt, ob eine passende Message empfangen wurde. Sonst hättest du ja keine Chance eine message an einen anderen Prozess zu schicken.

```suggestion
Each MessageStartEvent has to be configured with a Message.

A Message definition consists of an ID and a Name.
- The ID uniquely identifies the messages on the Process Model. 
- The name is used to actually send and receive the messages

When a message with a matching _name_ is received, a new Process Instance or the Process Model is created.
The Process Instance will start at the Message Event that has received the message.
```

> ![message_start_event](images/start_event/message_start_event.svg)

Reviewer: Hier sollte man das Start Event und am besten noch die im Property Panel hitnerlegte Config sehen.

The process defined above automatically starts, if the Message with the Message name

> _GoGoGo_ arrives.

Reviewer: ```suggestion
In this example, the Start Event listens for a message named `GoGoGo`.

When the AtlasEngine receives such a message, a new Process Instance will start from this Start Event.
```

## Signal Start Event

If you want to start your Process by a Received Signal, you can use the
SignalStartEvent.

The definition of this StartEvent variant is basically the same as in the
MessageStartEvent with the difference, that the Process will be started
at the SignalStartEvent, if a defined _Signal_ instead of a _Message_ arrives.

![signal_start_event](images/start_event/signal_start_event.svg)

You can see that this process automatically starts, if the Signal named

> _GoGoGo_ arrives.

Reviewer: Hier nochmal das gleiche wie beim `Message Start Event`.

## Timer Start Event

You can use the TimerStartEvent to start a Process time based.
This means that the StartEvent has an attached _TimerDefinition_ which

> trigger the StartEvent at different Times.

Reviewer: ```suggestion
`Timer Start Events` can be used to start Process Instances by use of a timer.
```

There are different TimerDefinitions which can be attached to the TimerStartEvent.
* **Date**: Its possible to set at Date which is in the future. If the given Date
  arrives, the StartEvent will be triggered.
* **Duration**: The StartEvent will be triggered after waiting a defined
  amount if time. This means that you can basically delay the actual start of
  your process.
* **Cycle**: Here, the StartEvent is triggered each time, the defined time
  interval is over. (It basically let the Process behave like a Cronjob, which

>   is executed repeatably.)

Reviewer: ```suggestion
There are three different types of timers that can be attached to a Start Event: 
- `Date`: When starting the Process Instance, it waits until the given date has passed
- `Duration`: When starting the Process Instance, it waits the given amount of time, before actually starting. Think of it as a time-delayed start
- `Cycle`: Uses a crontab to start new Process Instances from this Start Event periodically. 
Whenever the crontab expires, a new Process Instance is started automatically, starting at this Start Event
```

> ![timer_start_event](images/start_event/timer_start_event.svg)

Reviewer: Auch hier nur das Start Event mit Property Panel Config zeigen.

Here you can see that the process waits until tomorrow, until it actually is

> executed.

Reviewer: Woran erkenn ich das? 
Hier wäre es hilfreich, wenn man auch die Config im Property Panel sehen könnte.

Würd hier übrigens ein Beispiel mit Cronjob bringen, da das wohl das am häufigsten gebrauchte Szenario für Start Events ist.

# Start Event

A Start Event is used to mark an entry point for a Process Model.
Every Process Instance always starts at a Start Event.

> At least one StartEvent must exist on every Process Model, or it will not be executable.

Reviewer 2: ```suggestion
At least one Start Event must exist on every Process Model, or it will not be executable.
```

![Start Event](images/start_event/startevent_and_sequenceflow.svg)

There are several different types of Start Events:

## Regular Start Event

> Regular Start Events will just start the Process Instance, as it is described above.

Reviewer 2: Das hier würde ich noch einmal umformulieren.
```
as it is described above.
```
Ich kann weiter oben keine Erklärung finden die mir hier weiter hilft.

Reviewer: Ich würd den letzten Part auch einfach streichen.

```suggestion
Regular Start Events will just start the Process Instance.
```

### Properties

Each Start Event comes with two basic properties:
- `ID`: A unique identifier by which the Start Event can be identified throughout the Process Model.

> - `Name`: A descriptive name that you can use to describe what this Start Event does. Unlike the ID, it is not mandatory, is not bound to be unique and is also used as a display value in the BPMN Studio modeler

Reviewer: Das hier bitte unter den ersten Screenshot verschieben. Das hier gilt schließlich für alle Arten von Start Events.

Sollte am Ende so aussehen: 

```md

![Start Event](images/start_event/startevent_and_sequenceflow.svg)

## Properties

Each Start Event comes with two basic properties:
- `ID`: A unique identifier by which the Start Event can be identified throughout the Process Model.
- `Name`: A descriptive name that you can use to describe what this Start Event does. Unlike the ID, it is not mandatory, is not bound to be unique and is also used as a display value in the BPMN Studio modeler

## Types

There are several different types of Start Events:

### Multiple Start Events

A Process Model can have multiple Start Events. This can come in handy when you want to be able to start your process with varying prerequisites.

For example:

A backup process, that can be started by a cronjob on a `Timer Start Event` and a regular `Start Event` for starting the backup process manually.

> ![Multiple Start Events](images/start_event/multiple_start_events.svg)

Reviewer: Das hier sollte ans Ende, nachdem die Arten von Start Events vorgestellt wurden. 

Dann muss ich mich hier nicht erst fragen "Was ist denn ein Timer Start Event?", sondern habe die Erklärung dazu schonmal gelesen.

> ## Message Start Event

Reviewer: ```suggestion
### Message Start Event
```

This type of Start Event starts a process automatically, as soon as the message attached to the Start Event was received by the Atlas Engine.

Each Message Start Event has to be configured with a Message.

A Message definition consists of an ID and a Name.
- The ID uniquely identifies the messages on the Process Model
- The Name is used to actually send and receive the messages

![Message Start Event](images/start_event/startevent_message_gogogo.svg)
<img src="images/start_event/startevent_message_configuration.png" width="300px">

In this example, the Start Event listens for a message named `GoGoGo`.

When the Atlas Engine receives such a message, a new Process Instance will start from this Start Event.

> ## Signal Start Event

Reviewer: ```suggestion
### Signal Start Event
```

This type of Start Event starts a process automatically, as soon as the signal attached to the Start Event was received by the Atlas Engine.

The definition of this Start Event variant is basically the same as in the

> Message Start Event.

Reviewer: ```suggestion
Each Signal Start Event has to be configured with a Signal.
```

A Signal definition consists of an ID and a Name.
- The ID uniquely identifies the signals on the Process Model
- The Name is used to actually send and receive the signals

![Signal Start Event](images/start_event/signalstartevent_and_sequencflow.svg)
<img src="images/start_event/startevent_signal_configuration.png" width="300px">

In this example, the Start Event listens for a signal named `GoGoGo`.

When the Atlas Engine receives such a signal, a new Process Instance will start from this Start Event.

> ## Timer Start Event

Reviewer: ```suggestion
### Timer Start Event
```
